mobilecoind/src/service.rs

```diff
@@ -581,12 +603,31 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
         &mut self,
         request: mc_mobilecoind_api::CreateTransferCodeRequest,
     ) -> Result<mc_mobilecoind_api::CreateTransferCodeResponse, RpcStatus> {
-        if request.root_entropy.len() != 32 {
+        // Only allow one type of entropy.
+        if !request.bip39_entropy.is_empty() && !request.root_entropy.is_empty() {
             return Err(RpcStatus::new(
                 RpcStatusCode::INVALID_ARGUMENT,
-                Some("root_entropy".to_string()),
+                Some("bip39_entropy/root_entropy".to_string()),
             ));
         }
+
+        // If we were provided with bip39 entropy, ensure it is 32 bytes long.
+        if !request.bip39_entropy.is_empty() && request.bip39_entropy.len() != 32 {
```

Comment:
^ Related to the other comment about bip39 length.

Note: With the caveat that the mobilecoind's `create_transfer_code` could be more restrictive as to what it accepts for `bip39_entropy` and would still need to support transfer codes created elsewhere with less restrictions on the size of the entropy (unless we decide we want to restrict what is considered a valid `bip39_entropy` for transfer codes).

Reply:
I'm fine removing the restriction, instead I'll add a check that it can be converted to a `Mnemonic`.